dataservice/api/biospecimen/models.py

```diff
@@ -111,3 +131,108 @@ class Biospecimen(db.Model, Base):
     biospecimen_genomic_files = db.relationship(BiospecimenGenomicFile,
                                                 backref='biospecimen',
                                                 cascade='all, delete-orphan')
+    diagnoses = db.relationship('Diagnosis', secondary='biospecimen_diagnosis',
```

Comment:
Hmm I'm not I understand how this works or what it's doing. It kind of looks like an association proxy to Diagnosis? 
Do we need something like:
```
biospecimen_diagnoses = db.relationship('BiospecimenDiagnosis',
                                                  backref='Biospecimen',
                                                  cascade='all, delete-orphan')
```